armory/metrics/task.py

```diff
@@ -818,6 +820,139 @@ def object_detection_mAP(y_list, y_pred_list, iou_threshold=0.5, class_list=None
+             and error metrics that weight the significance of different types of errors to assess trade-offs between design choices for models
+             as well as potentially for attacks and defense
+
+    default max_dets=100 in TIDE
+    max_dets is an attribute for both data_ground_truth and data_detection
+    data_ground_truth.max_dets gets passed to tide.evaluate_range > TIDERun._run > TIDERun._eval_image > TIDEEXample._run (code snippet below)
+        preds = preds[:max_dets]
+        self.preds = preds # Update internally so TIDERun can update itself if :max_dets takes effect
+    for now assume we don't want max_dets to affect our metrics
+    set data_ground_truth.max_dets and data_detection.max_dets to max of max(len(y["labels"]), len(y_pred["labels"])))
+    after going through for loop
+    """
+
+    data_ground_truth_name = "ground_truth"
+    data_detection_name = "detection"
+
+    data_ground_truth = tidecv.data.Data(name=data_ground_truth_name)
+    data_detection = tidecv.data.Data(name=data_detection_name)
+
+    max_dets = 0
+    for i, (y, y_pred) in enumerate(zip(y_list, y_pred_list)):
+        if "image_id" in y:
+            image_id = y["image_id"][0]
+        else:
+            image_id = i
+
+        tidecv_ground_truth_list = armory_to_tide(y, image_id, is_detection=False)
+        for y_tidecv in tidecv_ground_truth_list:
+            data_ground_truth.add_ground_truth(**y_tidecv)
+
+        tidecv_detection_list = armory_to_tide(y_pred, image_id)
+        for y_tidecv in tidecv_detection_list:
+            data_detection.add_detection(**y_tidecv)
+
+        max_len = max(len(y["labels"]), len(y_pred["labels"]))
+        if max_len > max_dets:
+            max_dets = max_len
+
+    data_ground_truth.max_dets = max_dets
+    data_detection.max_dets = max_dets
+
+    tide = TIDE()
+    tide.evaluate_range(data_ground_truth, data_detection, mode=TIDE.BOX)
+    tide_error = tide.get_all_errors()
+    tide_error_count = {
+        k.short_name: len(v)
+        for k, v in tide.runs[data_detection.name].error_dict.items()
+    }
+    tide_fn_count = sum(
+        [len(v) for k, v in tide.runs[data_detection.name].false_negatives.items()]
+    )
+
+    armory_output = {
+        "mAP": {
+            x.pos_thresh: np.around(x.ap / 100, decimals=2)
+            for x in tide.run_thresholds[data_detection.name]
+        },
+        # not changing numeric format for error yet until we understand what the numbers mean
+        "errors": {
+            "main": {
+                "dAP": tide_error["main"][data_detection.name],
+                "count": tide_error_count,
+            },
+            "special": {
+                "dAP": tide_error["special"][data_detection.name],
+                "count": {"FalseNeg": tide_fn_count},
+            },
+        },
+    }
+
+    if return_tide:
+        return armory_output, tide
```

Comment:
Is this returning the `tide` object? If so, maybe update the kwarg to something like `return_tide_object`. That would make it clearer what's going on. Otherwise, it looks like it's asking whether you want to return the tide metrics.